### PR TITLE
Add coming soon template - split right image

### LINKED ISSUE #52979: Coming soon template 4
### Issue Description

P2: pfwcF5-13s-p2

Figma: 0BWMerqRR1f468EwC7g5tg-fi-3708_131070

Pre-requisite: #52393 and #52397

Image

### Acceptance Criteria

1. Build template using blocks
1. Ensure images and copy are editable

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #52979.

This PR adds a new Coming Soon template- split right image, built using WP blocks. The template follows the Figma design and ensures that all elements, including images and copy, are fully editable to provide flexibility for customization.

I had to add some custom CSS to make the template match the Figma design because Blocks Editor doesn't support it. For example, the cover block set text to center, but we want it to be top-aligned.

Screenshot 2024-12-17 at 11 45 44

Screenshot 2024-12-17 at 11 46 49

Screenshot 2024-12-17 at 11 54 21

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site with  `coming-soon-newsletter-template` feature flag enabled.
2. Enable coming soon mode in site visibility settings.
3. Go to `Appearance > Editor > Templates > Coming soon`
4. Choose `Coming Soon Split Right Image` template on the right sidebar (Template > Design > Coming Soon Split Right Image)
6. Confirm the template matches the design in Figma.
7. Confirm that you can customize the styles, replace image, change the content, etc.
8. Select `Coming Soon Split Right Image` template again and save the template.
9. Go to the site front e incognito tab to make sure the template is loaded correctly. The page should be responsive.
10. Go to `Appearance > Editor > Templates` and confirm the coming soon template preview shows the selected template.

Figma:  0BWMerqRR1f468EwC7g5tg-fi-3708_131070

image

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Add coming soon template - split right image

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/assets/images/pattern-placeholders/orange-wall-with-bicycle.jpg
Comment (ilyasfoo) [inline plugins/woocommerce/assets/images/pattern-placeholders/orange-wall-with-bicycle.jpg:1]: Compressed down to 38kb:

https://github.com/user-attachments/assets/eafb07a7-00cc-4c3b-9665-fc750e7c2054

File: plugins/woocommerce/assets/images/pattern-placeholders/wheel-leaf-bicycle-bike-vehicle-spoke.jpg
Comment (ilyasfoo) [inline plugins/woocommerce/assets/images/pattern-placeholders/wheel-leaf-bicycle-bike-vehicle-spoke.jpg:1]: Compressed down to 7kb:

https://github.com/user-attachments/assets/3c26c6d9-5c8b-4060-87e7-fb30d4bfd710

File: plugins/woocommerce/patterns/page-coming-soon-split-right-image.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+				<div class="wp-block-cover is-light woocommerce-split-right-image-cover" style="padding-top:126px;padding-right:75px;padding-bottom:126px;padding-left:76px;min-height:100%"><span aria-hidden="true" class="wp-block-cover__background has-background-dim-0 has-background-dim" style="background-color:#cd7550"></span><img class="wp-block-cover__image-background wp-image-34" alt="" src="<?php echo esc_url( $right_image ); ?>" style="object-position:50% 50%" data-object-fit="cover" data-object-position="50% 50%"/><div class="wp-block-cover__inner-container"><!-- wp:paragraph {"align":"center","placeholder":"Write title…","style":{"typography":{"fontSize":"75px","fontStyle":"normal","fontWeight":"400","lineHeight":"1.3"},"color":{"text":"#ffffff80"},"elements":{"link":{"color":{"text":"#ffffff80"}}}},"fontFamily":"<?php echo esc_attr( $cardo_font_family ); ?>"} -->
```
Comment (ilyasfoo) [inline plugins/woocommerce/patterns/page-coming-soon-split-right-image.php]: I find this to be better, especially on smaller screens where the bike will be in focus

image

```suggestion
			<div class="wp-block-column is-vertically-aligned-stretch"><!-- wp:cover {"url":"<?php echo esc_url( $right_image ); ?>","id":34,"dimRatio":0,"customOverlayColor":"#cd7550","isUserOverlayColor":false,"focalPoint":{"x":0.5,"y":0.8},"minHeight":100,"minHeightUnit":"%","isDark":false,"className":"woocommerce-split-right-image-cover","style":{"spacing":{"padding":{"top":"126px","bottom":"126px","left":"76px","right":"75px"}}},"layout":{"type":"default"}} -->
				<div class="wp-block-cover is-light woocommerce-split-right-image-cover" style="padding-top:126px;padding-right:75px;padding-bottom:126px;padding-left:76px;min-height:100%"><span aria-hidden="true" class="wp-block-cover__background has-background-dim-0 has-background-dim" style="background-color:#cd7550"></span><img class="wp-block-cover__image-background wp-image-34" alt="" src="<?php echo esc_url( $right_image ); ?>" style="object-position:50% 80%" data-object-fit="cover" data-object-position="50% 80%"/><div class="wp-block-cover__inner-container"><!-- wp:paragraph {"align":"center","placeholder":"Write title…","style":{"typography":{"fontSize":"75px","fontStyle":"normal","fontWeight":"400","lineHeight":"1.3"},"color":{"text":"#ffffff80"},"elements":{"link":{"color":{"text":"#ffffff80"}}}},"fontFamily":"<?php echo esc_attr( $cardo_font_family ); ?>"} -->
```

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
Looking great, @chihsuan! I have some minor comments.

I also helped compressed images using https://compressor.io/ where I think we can have significant savings. Visually they look the same to me, but let me know if you find it to be of low quality.

**Comment by chihsuan** [comment]
> I also helped compressed images using https://compressor.io/ where I think we can have significant savings. Visually they look the same to me, but let me know if you find it to be of low quality.

Thanks for the help! @ilyasfoo This is looking great. Updated the images in 2534e388d7d8e168595f130d8df062c5fa96880b.

**Review by ilyasfoo — APPROVED** [review]
Awesome, thanks! Tests well, LGTM!

**Review by rjchow — APPROVED** [review]
Tested well and looks good! Nice catch on the image optimisation @ilyasfoo !

**Pull request merged**